lib/errors.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+/**
+ * @module Errors
+ */
+/**
+ * Factory functions to create throwable error objects
+ */
+
+/**
+ * @summary
+ * File/s of test could not be found.
+ *
+ * @public
+ */
+function NoFilesMatchPatternError(message, pattern) {
```

Comment:
Assuming we're not going to use custom constructors and instead we're just going to be assigning properties on the `Error` object... I would prefer we don't add *additional* properties (e.g., `pattern`, `argument`) to the error object, for sake of simplicity.  We're probably not going to need them.  Let's just stuff these in the `message` and call it good?  What's the use case?

This contradicts what @plroebuck commented on, so would like to understand why you and he feel this is needed.

Reply:
The whole point of using error codes was to stop having to parse the error message. Only by adding additional properties (as well as the code) does i18n/l10n become possible. Not to mention being easier to use programmatically. This style is used heavily by Node itself already for [system errors](https://nodejs.org/api/errors.html#errors_class_systemerror)

Reply:
Alright.  I feel it's kind of a "YAGNI" situation, but won't die on that hill.